iree/compiler/Dialect/HAL/IR/HALOps.td

```diff
@@ -1841,14 +1841,35 @@ def HAL_ExecutableEndOp : HAL_Op<"executable_end", [
   let assemblyFormat = "attr-dict";
 }
 
+def HAL_YieldOp : HAL_Op<"yield", [NoSideEffect, Terminator]> {
```

Comment:
there's already a hal.return op - let's use that instead (or, change everything to use hal.yield if that's more canonical - I think flow.return/hal.return predate the yield terminology being used upstream - but not both)